File: Front-end/Front_end/src/types/trip.ts
export interface Expense {
  id: string;
  category: string;   // 식비, 교통, 숙박 등
  amount: number;
  time: string;       // ISO string
}

export interface Schedule {
  id: string;
  time: string;       // ISO string (예: 2025-09-01T09:00)
  description: string;
  category: string;
  expectedCost: number;
  actualCost?: number;
}

export interface Trip {
  id: string;
  title: string;
  startDate: string;
  endDate: string;
  totalBudget: number;
  schedules: Schedule[];
  expenses: Expense[];
}

File: rare/src/main/java/com/jinoh/rareitemServer/domain/Trip.java
package com.jinoh.rareitemServer.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private LocalDate startDate; // "2025-09-01" 형태

    private LocalDate endDate;

    private int totalBudget;

    // 일정(1:N)
    @OneToMany(mappedBy = "trip", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @JsonManagedReference // 순환참조 방지 (아래 Schedule의 @JsonBackReference와 세트)
    private List<Schedule> schedules = new ArrayList<>();



}
